### PR TITLE
taylors-dec12-weekend-hacking

### PR DESCRIPTION
**Pull request by defuse**

### CODE REVIEW

File: test/unit/FileTest.php [hunk 1]
```diff
@@ -106,15 +105,65 @@ public function testResourceToResource($srcFile)
...
+        file_put_contents($junk, "");
+        File::decryptFile($junk, self::$TEMP_DIR . '/unjunked', $this->key);
+    }
+
+    /**
+     * @expectedException \Defuse\Crypto\Exception\InvalidCiphertextException
+     */
+    public function testDecryptTruncatedCiphertext()
+    {
+        // This tests for issue #115 on GitHub.
```
Comment (sstok) [inline test/unit/FileTest.php]: Better to add a small title to give idea - `Malformed ciphertext files can cause `fread` with negative values upon decryption`

**Comment by paragonie-scott** [comment]
:+1: though I do agree with @sstok's suggestion

**Comment by defuse** [comment]
Uh oh I'm so confused as to what code is what now.

**Comment by defuse** [comment]
Oh FFS why didn't I merge this on dec 12. I shot myself in the foot, there are going to be so many merge conflicts.

**Pull request merged**